test/test_train_mp_imagenet.py

```diff
@@ -301,16 +301,17 @@ def train_loop_fn(loader, epoch):
           output = model(data)
           loss = loss_fn(output, target)
           loss.backward()
-      if FLAGS.ddp:
-        optimizer.step()
-      else:
-        xm.optimizer_step(optimizer)
-      tracker.add(FLAGS.batch_size)
-      if lr_scheduler:
-        lr_scheduler.step()
-      if step % FLAGS.log_steps == 0:
-        xm.add_step_closure(
-            _train_update, args=(device, step, loss, tracker, epoch, writer))
+          if FLAGS.ddp:
+            optimizer.step()
+          else:
+            xm.optimizer_step(optimizer)
+            tracker.add(FLAGS.batch_size)
+          if lr_scheduler:
+            lr_scheduler.step()
+          if step % FLAGS.log_steps == 0:
+            xm.add_step_closure(
```

Comment:
Let's move `step_ckousre` outside of `build_graph`, it is not actually building any graph. I felt like same way about the `lr_scheduler`. I think including `optimizer` make sense.

Reply:
Done.